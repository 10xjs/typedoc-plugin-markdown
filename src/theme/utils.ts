import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
