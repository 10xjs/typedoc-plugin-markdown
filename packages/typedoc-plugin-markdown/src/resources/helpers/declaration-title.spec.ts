import * as Handlebars from 'handlebars';

import { TestApp } from '../../../test/utils';

describe(`[helpers] declarationTitle`, () => {
  let testApp: TestApp;
  beforeAll(() => {
    testApp = new TestApp(['variables']);
    testApp.bootstrap();
  });

  test(`should compile`, () => {
    expect(
      Handlebars.helpers.declarationTitle.call(
        testApp.project.findReflectionByName('color'),
      ),
    ).toMatchSnapshot();
  });
});
