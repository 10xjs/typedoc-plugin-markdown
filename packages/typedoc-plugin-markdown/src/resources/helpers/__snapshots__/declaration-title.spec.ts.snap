// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[helpers] declarationTitle should compile 1`] = `"\`Const\`  **color**: \\"blue\\"  = \\"blue\\""`;
